app/org/sagebionetworks/bridge/services/AccountWorkflowService.java

```diff
@@ -225,17 +227,45 @@ public void sendEmailVerificationToken(Study study, String userId, String recipi
         sendMailService.sendEmail(provider);
     }
     
+    public void sendPhoneVerificationToken(Study study, String userId, Phone phone) {
+        checkNotNull(study);
+        checkArgument(isNotBlank(userId));
+        
+        if (phone == null) {
+            return;
+        }
+        if (isRequestThrottled(ThrottleRequestType.VERIFY_PHONE, userId)) {
+            // Too many requests. Throttle.
+            return;
+        }
+        String sptoken = getNextToken();
+        String formattedSpToken = sptoken.substring(0,3) + "-" + sptoken.substring(3,6);
```

Comment:
Added that, added verification of the cache call in sendEmailVerificationToken and sendPhoneVerificationToken tests (because that's obviously missing). Added the channel type to the verifyChannel method, and now check it on verification. Added tests to verify this is enforced.